ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/Util/AcidState.hs

```diff
@@ -101,3 +108,69 @@ zoomAll l upd = StateT $ \large -> do
 mustBeRight :: Either Void b -> b
 mustBeRight (Left  a) = absurd a
 mustBeRight (Right b) = b
+
+type MonadAcidCleanup m =
+    ( MonadIO m
+    , MonadMask m
+    )
+
+
+-- | Helper function to run action periodically.
+runPeriodically :: forall (unit :: Rat) m a. (KnownDivRat unit Second, MonadIO m)
+  => Time unit -- ^ time between performing action
+  -> m a       -- ^ action
+  -> m ()
+runPeriodically delay action = forever $ do
+  _ <- action
+  threadDelay delay
+
+
+-- | Creates checkpoint of the current state of Acid DB,
+-- archive previous checkpoints and remove too old archives.
+-- This is needed to avoid storing all logs permanently.
+cleanupAcidState ::
+       forall m st. (MonadAcidCleanup m)
+    => AcidState st
+    -> FilePath
+    -> Int
+    -> (Severity -> Text -> m ())
+    -> m ()
+cleanupAcidState db path numberOfStoredArchives writeLog = perform
+  where
+    perform = cleanupAction `catchAny` handler
```

Comment:
Is there any need to make `perform` a separate function? If not, could you change it to the following:

```haskell
cleanupAcidState db path numberOfStoredArchives writeLog =
    cleanupAction `catchAny` handler
  where
    ...
```

Reply:
I use the whole function in the handler inside it and I use "perform" in it just to be short.